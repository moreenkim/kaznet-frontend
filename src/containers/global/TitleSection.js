// Smart component that renders the page title section
import React, { Component } from "react";
import { connect } from "react-redux";

import * as globalSelectors from "../../store/global/reducer";

import PageTitle from "../../components/page/PageTitle";
import DetailPageTitle from "../../components/page/DetailTitle";

export class TitleSection extends Component {
  render() {
    if (!globalSelectors.getNoTitle) {
      if (!globalSelectors.getShowDetail) {
        return (
          <PageTitle
            pageTitle={this.props.pageTitle}
            pageTitleButton={this.props.pageTitleButton}
            pageTarget={this.props.pageTarget}
          />
        );
      } else {
        return <DetailPageTitle />;
      }
    }
    return null;
  }
}

// which props do we want to inject, given the global store state?
function mapStateToProps(state) {
  return {
    pageTitle: globalSelectors.getPageTitle(state),
    pageTitleButton: globalSelectors.getPageTitleButton(state),
    pageTarget: globalSelectors.getPageTarget(state)
  };
}

export default connect(mapStateToProps)(TitleSection);
