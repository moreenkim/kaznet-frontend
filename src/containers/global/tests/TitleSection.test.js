// Test TitleSection
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { TitleSection } from "../TitleSection";

describe("containers/global/TitleSection", () => {
  it("renders without crashing", () => {
    shallow(<TitleSection />).dive();
  });

  it("renders title section correctly", () => {
    const wrapper = mount(<TitleSection />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
