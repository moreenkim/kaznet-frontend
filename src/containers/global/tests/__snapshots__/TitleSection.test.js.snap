// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/global/TitleSection renders detail title correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <TitleSection
    pageTarget="/"
    showDetail={true}
    showTitle={true}
  >
    <DetailPageTitle
      pageTarget="/"
    >
      <section
        className="detail-page-title"
      >
        <Col
          md={
            Object {
              "offset": 2,
              "size": 8,
            }
          }
          sm="12"
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <div
            className="col-sm-12 col-md-8 offset-md-2"
          >
            <div>
              <h1
                className="kaznet-detail-title"
              >
                <Link
                  className="kaznet-header-link"
                  replace={false}
                  to="/"
                >
                  <a
                    className="kaznet-header-link"
                    href="/"
                    onClick={[Function]}
                  />
                </Link>
                 &gt; 
              </h1>
              <p
                className="kaznet-creation-detail"
              >
                By Jensen Nathan, 21 May 2018
              </p>
              <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <div
                  className="col-md-12"
                >
                  <Row
                    className="kaznet-action-links"
                    tag="div"
                  >
                    <div
                      className="kaznet-action-links row"
                    >
                      <Link
                        className="action-link archive-button"
                        replace={false}
                        to="/"
                      >
                        <a
                          className="action-link archive-button"
                          href="/"
                          onClick={[Function]}
                        >
                          <FontAwesomeIcon
                            border={false}
                            className="withspace"
                            fixedWidth={false}
                            flip={null}
                            icon="folder-open"
                            listItem={false}
                            mask={null}
                            name=""
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-folder-open fa-w-18 withspace"
                              data-icon="folder-open"
                              data-prefix="fas"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 576 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M572.694 292.093L500.27 416.248A63.997 63.997 0 0 1 444.989 448H45.025c-18.523 0-30.064-20.093-20.731-36.093l72.424-124.155A64 64 0 0 1 152 256h399.964c18.523 0 30.064 20.093 20.73 36.093zM152 224h328v-48c0-26.51-21.49-48-48-48H272l-64-64H48C21.49 64 0 85.49 0 112v278.046l69.077-118.418C86.214 242.25 117.989 224 152 224z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                          Archive
                        </a>
                      </Link>
                    </div>
                  </Row>
                </div>
              </Col>
            </div>
          </div>
        </Col>
      </section>
    </DetailPageTitle>
  </TitleSection>
</Router>
`;

exports[`containers/global/TitleSection renders page title correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <TitleSection
    showDetail={false}
    showTitle={true}
  >
    <PageTitle>
      <section
        className="page-title"
      >
        <Row
          tag="div"
        >
          <div
            className="row"
          >
            <Col
              md={
                Object {
                  "offset": 2,
                  "size": 8,
                }
              }
              sm="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <div
                className="col-sm-12 col-md-8 offset-md-2"
              >
                <div>
                  <h1 />
                  <Row
                    tag="div"
                  >
                    <div
                      className="row"
                    >
                      <Col
                        md="12"
                        tag="div"
                        widths={
                          Array [
                            "xs",
                            "sm",
                            "md",
                            "lg",
                            "xl",
                          ]
                        }
                      >
                        <div
                          className="col-md-12"
                        >
                          <Form
                            tag="form"
                          >
                            <form
                              className=""
                            >
                              <InputGroup
                                className="search-group"
                                tag="div"
                              >
                                <div
                                  className="search-group input-group"
                                >
                                  <InputGroupAddon
                                    addonType="prepend"
                                    className="search-prepend"
                                    tag="div"
                                  >
                                    <div
                                      className="search-prepend input-group-prepend"
                                    >
                                      <InputGroupText
                                        className="bg-white border-right-0"
                                        tag="span"
                                      >
                                        <span
                                          className="bg-white border-right-0 input-group-text"
                                        >
                                          <Button
                                            aria-label="Filter"
                                            className="search_button btn btn-deoco"
                                            color="deoco"
                                            tag="button"
                                            type="submit"
                                          >
                                            <button
                                              aria-label="Filter"
                                              className="search_button btn btn-deoco btn btn-deoco"
                                              onClick={[Function]}
                                              type="submit"
                                            >
                                              <FontAwesomeIcon
                                                border={false}
                                                className="filtersubmit"
                                                fixedWidth={false}
                                                flip={null}
                                                icon="search"
                                                listItem={false}
                                                mask={null}
                                                name=""
                                                pull={null}
                                                pulse={false}
                                                rotation={null}
                                                size={null}
                                                spin={false}
                                                symbol={false}
                                                transform={null}
                                              >
                                                <svg
                                                  aria-hidden="true"
                                                  className="svg-inline--fa fa-search fa-w-16 filtersubmit"
                                                  data-icon="search"
                                                  data-prefix="fas"
                                                  role="img"
                                                  style={Object {}}
                                                  viewBox="0 0 512 512"
                                                  xmlns="http://www.w3.org/2000/svg"
                                                >
                                                  <path
                                                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                                                    fill="currentColor"
                                                    style={Object {}}
                                                  />
                                                </svg>
                                              </FontAwesomeIcon>
                                            </button>
                                          </Button>
                                        </span>
                                      </InputGroupText>
                                    </div>
                                  </InputGroupAddon>
                                  <Input
                                    aria-label="Search"
                                    bsSize="lg"
                                    className="border-left-0"
                                    defaultValue=""
                                    name="search"
                                    placeholder="Search"
                                    type="text"
                                  >
                                    <input
                                      aria-label="Search"
                                      className="border-left-0 form-control-lg form-control"
                                      defaultValue=""
                                      name="search"
                                      placeholder="Search"
                                      type="text"
                                    />
                                  </Input>
                                </div>
                              </InputGroup>
                            </form>
                          </Form>
                        </div>
                      </Col>
                    </div>
                  </Row>
                </div>
              </div>
            </Col>
          </div>
        </Row>
      </section>
    </PageTitle>
  </TitleSection>
</Router>
`;
